pkg/provisioner/ironic/raid.go

```diff
@@ -99,11 +99,26 @@ func buildTargetHardwareRAIDCfg(volumes []metal3v1alpha1.HardwareRAIDVolume) (lo
 			}
 			nameCheckFlags[volume.Name] = index
 		}
+
+		// Check numberOfPhysicalDisks is same as len(physicalDisks)
+		if volume.NumberOfPhysicalDisks != nil {
+			if *volume.NumberOfPhysicalDisks != len(volume.PhysicalDisks) {
```

Comment:
This effectively makes volume.PhysicalDisks a required property doesn't it? I think we should only compare if they are both non-zero.

Reply:
ah! good catch! i'll fix it right away.